hugolib/page.go

```diff
@@ -1021,6 +1021,14 @@ func (p *Page) update(f interface{}) error {
 			p.Date = fi.ModTime()
 			p.Params["date"] = p.Date
 		}
+	} else if p.Date.IsZero() && p.s.Cfg.GetBool("useFilenameDateAsFallback") {
```

Comment:
I'm thinking so - its balance between config and 'magic'.

If the regex and time format is always the same, you can infer it from named groups.

For example:
```
(?P<datestamp>(?P<stdYear>\d{4})\-(?P<stdZeroMonth>\d{2})-(?P<stdZeroDay>\d{2}))
```

The catch is that time format const's aren't public so there would need to be a copy of them in Hugo for mapping (probably just day/month/year with the 0 padded variants too).

So its either:
 a) add 2x config params and let people choose format and pattern
 b) add 1x config param and force people to use the same format as the pattern.